### ISSUE TITLE
SpawnActorPower: cannot filter for Terrain

### ISSUE THREAD

**Issue by MlemandPurrs** [OpenRA/OpenRA#17371]
An SpawnActorPower can spawn the given unit on spaces the unit given normally wouldnt able to go to. This can potentially cause Vehicles on Water, Buildings on Water, inside Cliffs, etc. There should be an way to limit this, like it would check if space is buildable then spawn the building there, or check if space is passable for unit's movetype and block if uneligible.

**Comment by pchote** [comment]
If you are using this for ejecting pilots on unit death, the `EjectOnDeath` trait has an option to disable spawning if the pilot can't exist in the cell.

**Comment by MlemandPurrs** [comment]
I am using this for:
```
	SpawnActorPower@LoneTower:
		OrderName: SovietLoneTowerPowerInfoOrder
		Prerequisites: ~structures.soviet, player.uzbek
		Actor: ftur.uzbek
		LifeTime: -1
		ChargeInterval: 7500
		Icon: fturicon
		Description: Lone Tower
		LongDesc: A Flame Tower gets instantly constructed on the spot\nsadly not checking passability and build-radius.
		SelectTargetSpeechNotification: SelectTarget
		LaunchSpeechNotification: ReinforcementsArrived
		PauseOnCondition: lowpower || disabled
```

**Comment by GraionDilach** [comment]
Downstream has `SpawnBuildingWarhead` for this usecase (should have been merged into downstream `SpawnActorWarhead` but footprint handling would be too complicated then).

https://github.com/AttacqueSuperior/Engine/tree/master/OpenRA.Mods.AS/Warheads
